cdisc_rules_engine/services/data_readers/data_reader_factory.py

```diff
@@ -1,7 +1,39 @@
+from typing import Type
+
+from cdisc_rules_engine.services.data_readers.data_reader_interface import (
+    DataReaderInterface,
+)
 from cdisc_rules_engine.services.data_readers.xpt_reader import XPTReader
+from cdisc_rules_engine.services.factory_interface import FactoryInterface
+
+
+class DataReaderFactory(FactoryInterface):
+    _service_map = {
+        "xpt": XPTReader,
+    }
+
+    def __init__(self):
+        self._default_service_name: str = "xpt"
 
+    @classmethod
+    def register_service(cls, name: str, service: Type[DataReaderInterface]):
```

Comment:
These methods should probably be called register reader and get_reader. Also the _service_map should be reader_map

Reply:
We tend to make all factories implement the same interface for convenience. The interface defines the methods as register_service, get_service. Good catch about reader_map, will do.

Reply:
Sounds good. I will update the operations factory to use get_service, and register_service method

Reply:
Yeah, just inherit them from interface